velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -235,23 +304,35 @@ void WindowFuzzer::go() {
     const auto call =
         makeFunctionCall(signature.name, argNames, false, false, ignoreNulls);
 
+    // Randomly select if ROWS or RANGE frame
+    auto frameType = vectorFuzzer_.coinToss(0.5)
+        ? core::WindowNode::WindowType::kRows
+        : core::WindowNode::WindowType::kRange;
+
     std::vector<SortingKeyAndOrder> sortingKeysAndOrders;
     // 50% chance without order-by clause.
     if (vectorFuzzer_.coinToss(0.5)) {
-      sortingKeysAndOrders =
-          generateSortingKeysAndOrders("s", argNames, argTypes);
+      sortingKeysAndOrders = generateSortingKeysAndOrders(
+          "s",
+          argNames,
+          argTypes,
+          frameType == core::WindowNode::WindowType::kRange);
```

Comment:
Hi @minhancao, `frameType == WindowType::kRange` doesn't necessarily mean it is a k-range frame because BoundType could be kUnboundedPreceding, kUnboundedFollowing, or kCurrentRow that doesn't require single order-by key. I wonder whether we could determine the BoundType before calling generateSortingKeysAndOrders() as well so that we know both the WindowType and BoundType. 

I saw that BoundType is currently determined in generateFrameClause() that expects `input` as an argument because it adds the k-range frame columns to `input`. Since this method is pretty big, I wonder if we could split generateFrameClause() into 3 smaller methods? For example,
1. Determine the BoundTypes of the frame. We can call generateSortingKeysAndOrders() after it.
2. Generate the frame string given the WindowType and BoundTypes, e.g., "RANGE BETWEEN k0 PRECEDING and k1 FOLLOWING".
3. Add the k-range frame column to input given the WindowType and BoundType.

What do you think?

Reply:
Agree with Wei. We need to use both range and bound type to determine the number of sorting keys.

Reply:
I agree as well, will implement your suggestion